.github/workflows/bash_unit_testing.yml

```diff
@@ -3,8 +3,9 @@ name: bash Unit Testing
 on:
   pull_request:
 
+# Unique to this workflow and PR, or SHA if not a PR
 concurrency:
-  group: bash-unit-test-${{ github.head_ref  || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
```

Comment:
The docs for `run_id` say
> A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run.

Which means if a new commit is pushed, the runs for the previous commit are not cancelled because the `group` here is different for each commit.